### ISSUE TITLE
Swin Transformer: Hierarchical Vision Transformer using Shifted Windows

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#6]
- 새로운 vision transformer인 swin transformer 제안.
- 기존에 NLP (natural language processing)에서 좋은 성능을 낸 transformer를 computer vision task에도 쓸 수 있도록 general-purpose backbone으로 만듦.

![image](https://user-images.githubusercontent.com/67627471/127636047-29d42372-5542-4a59-8cd4-3bdd7c4a4db0.png)

![image](https://user-images.githubusercontent.com/67627471/127636263-9f551c2c-8a0b-4aae-a6a4-aa9825bef156.png)

- Patch Partition에서 RGB 이미지를 non-overlapping patches로 나눔. 이렇게 나눠진 path는 token으로 처리. (기존 transformer는 고정된 크기의 token을 쓰지만 이 논문에서는 계층적 representation을 씀. => 첫 번째 이미지에서 Layer 1의 gray patch에서 read windows로 deeper layer에서 merge하는 방식.)
- patch들을 linear embedding layer에서 arbitrary dimension에 project함.
- 다음에 patch merging layer에서 2x2의 이웃한 patch들끼리 features를 concatenate함.
- 그 다음 Swing Transformer Block에서 feature transformation 수행함.
- (b)에서 W-MSA와 SW-MSA는  self-attention으로 앞뒤로 Layer normalization을 해줌. 그런 다음 마지막에 MLP(multi-layer perceptron)을 적용해준 뒤 successive block 인풋으로 넘김. 중간에 2번의 residual learning을 거침. MLP에서는 GELU (비선형) 활성화 함수를 씀.
- window 기반의 self-attention은 다른 윈도우끼리의 connection이 떨어지는데, 이 논문에서는 shifted window를 써서 처음 4분할 윈도우로 partition한 다음에 shift된 윈도우로 다시 partition해서 connection을 introduce해줌.
- shifted partitioning하면 윈도우 수가 많아져서 복잡도가 높아진다는 단점이 있음. 이를 해결하기 위해 cyclic-shifting을 적용함. 한 윈도우가 인접하지 않은 여러 개의 서브 윈도우로 구성되게 됨.
- 절대적인 position embedding을 집어넣으면 성능이 떨어지므로 적용하지 않았음. 다만 pre-train된 상대적인 position bias를 model의 fine-tuning에 쓰일 수는 있음.
- 성능:
- (1) image classification (86.4 top-1 accuracy on ImageNet-1K)
- (2) dense prediction tasks such as object detection (58.7 box AP and 51.1 mask AP on COCO test-dev)
- (3) semantic segmentation (53.5 mIoU on ADE20K val)
- 기존 SOA 성능을 뛰어넘으면서 CV 백본으로써 transformer 기반 모델의 가능성을 보여줌.
